src/components/NftCard/NftCard.tsx

```diff
@@ -1,23 +1,29 @@
 import { useWallet } from '@solana/wallet-adapter-react'
 import { PublicKey } from '@solana/web3.js'
-import { equals, find, not, pipe, prop, when, isNil, always } from 'ramda'
+import { equals, find, not, pipe, prop, when, isNil, always, map } from 'ramda'
 import React from 'react'
 import Link from 'next/link'
 import { addressAvatar } from 'src/modules/address'
-import { toSOL } from './../../modules/lamports'
-import { Listing, Marketplace, Nft } from './../../types'
+import { Listing, Marketplace, Nft } from '@holaplex/marketplace-js-sdk'
+import Price from '../Price'
+import { TokenInfo } from '@solana/spl-token-registry'
 
 interface NftCardProps {
   nft: Nft
   marketplace: Marketplace
+  tokenMap: Map<string, TokenInfo>
 }
 
-export const NftCard = ({ nft, marketplace }: NftCardProps) => {
+export const NftCard = ({ nft, marketplace, tokenMap }: NftCardProps) => {
   const { publicKey } = useWallet()
-  const listing = find<Listing>(
-    pipe(prop('auctionHouse'), equals(marketplace.auctionHouse.address))
-  )(nft.listings)
-
+  const listing = !nft.listings
+    ? null
+    : find<Listing>(
+        pipe(
+          prop('auctionHouse.address'),
+          equals(map(prop('address'))(marketplace.auctionHouses))
+        )
+      )(nft.listings)
```

Comment:
```suggestion
  const listing = when(
    pipe(isEmpty, not),
     find<Listing>(
        pipe(
          prop('auctionHouse.address'),
          equals(map(prop('address'))(marketplace.auctionHouses))
        )
  )(nft.listings)
```

when performs transform if condidition is met otherwise returns the value. In this case, transform listings if they aren't empty otherwise return the empty list.

If you need null can use `ifElse(isEmpty, always(null), ...)`